conans/client/installer.py

```diff
@@ -36,76 +36,225 @@ def _init_package_info(deps_graph, paths, current_path):
         conan_file.user_info = UserInfo()
 
 
-def build_id(conanfile):
-    if hasattr(conanfile, "build_id"):
+def build_id(conan_file):
+    if hasattr(conan_file, "build_id"):
         # construct new ConanInfo
-        build_id_info = conanfile.info.copy()
-        conanfile.info_build = build_id_info
+        build_id_info = conan_file.info.copy()
+        conan_file.info_build = build_id_info
         # effectively call the user function to change the package values
-        with conanfile_exception_formatter(str(conanfile), "build_id"):
-            conanfile.build_id()
+        with conanfile_exception_formatter(str(conan_file), "build_id"):
+            conan_file.build_id()
         # compute modified ID
         return build_id_info.package_id()
     return None
 
 
-class BuildMode(object):
-    def __init__(self, params, output):
+class _ConanPackageBuilder(object):
```

Comment:
Private class (only used by installer) to build and package a single binary package.